helix-core/src/main/java/org/apache/helix/manager/zk/ZkBucketDataAccessor.java

```diff
@@ -156,10 +165,9 @@ public <T extends HelixProperty> boolean compressedBucketWrite(String rootPath,
       lastWriteVersion++;
       return String.valueOf(lastWriteVersion).getBytes();
     };
-
     // 1. Increment lastWriteVersion using DataUpdater
     ZkBaseDataAccessor.AccessResult result = _zkBaseDataAccessor.doUpdate(
-        rootPath + "/" + LAST_WRITE_KEY, lastWriteVersionUpdater, AccessOption.PERSISTENT);
```

Comment:
based on how pr dicussion concludes, we might need to use explicit create for first time and then update calls from second time onwards.